src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        joinString += '';
+      } else {
+        joinString += this[i];
+      }
+
+      if (i !== this.length - 1) {
```

Comment:
I also suggest putting this.length in a variable. since you're using it twice already, but I guess it's optional